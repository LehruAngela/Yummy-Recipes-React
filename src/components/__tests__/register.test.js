import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Register from '../register';

describe('Register', () => {
    const wrapper = shallow(<Register />);
    const preventDefault = jest.fn();

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
      }); 

    it('handles Register', () => {
        expect(wrapper.instance().handleRegister({ preventDefault }));
    });

 
});