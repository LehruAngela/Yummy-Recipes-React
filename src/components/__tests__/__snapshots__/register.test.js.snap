// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register renders properly 1`] = `
<div
  className="form-margin"
>
  <h1>
    Yummy Recipes App
  </h1>
  <br />
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <h4>
        <label
          htmlFor="email"
        >
          Email:
        </label>
      </h4>
      <input
        name="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <h4>
        <label
          htmlFor="text"
        >
          Username:
        </label>
        <br />
      </h4>
      <input
        name="username"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <h4>
        <label
          htmlFor="password"
        >
          Password:
        </label>
        <br />
      </h4>
      <input
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <h4>
        <label
          htmlFor="confirmpassword"
        >
          Confirm Password:
        </label>
        <br />
      </h4>
      <input
        name="confirm_password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <button
        className="btn btn-outline-warning"
        type="submit"
      >
        Register
      </button>
    </div>
    <br />
  </form>
  <h4>
    Already have an account?
  </h4>
  <div>
    <Link
      replace={false}
      to="/login"
    >
      <button
        className="btn btn-outline-warning"
        type="submit"
      >
        Login
      </button>
    </Link>
  </div>
  <br />
  <br />
  <footer>
    <p
      className="copyright text-muted small footer"
    >
      Copyright Â© Gela 2018.
    </p>
  </footer>
</div>
`;
